engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -600,6 +605,19 @@ public void activate()
       CHECKPOINT_WINDOW_COUNT = 1;
     }
 
+    int dagChkptWndwCnt = context.getValue(Context.DAGContext.CHECKPOINT_WINDOW_COUNT);
+    if (PROCESSING_MODE != ProcessingMode.EXACTLY_ONCE) {
+      int chkOffset = dagChkptWndwCnt % CHECKPOINT_WINDOW_COUNT;
+      if (chkOffset != 0) {
+        EFFECTIVE_CHECKPOINT_WINDOW_COUNT = dagChkptWndwCnt + CHECKPOINT_WINDOW_COUNT - chkOffset;
+      } else {
+        EFFECTIVE_CHECKPOINT_WINDOW_COUNT = dagChkptWndwCnt;
+      }
+    } else {
+      EFFECTIVE_CHECKPOINT_WINDOW_COUNT = 1;
+    }
```

Comment:
@chandnisingh This is for iteration support, APEX-246. We need to know at the beginning of the window before the checkpoint. Initial iteration support will be for a delay of 1 but that could change in future to be more than 1 window. This is to let you know how far you are from the checkpoint window.

@chandnisingh, @gauravgopi123 The CHECKPOINT_WINDOW_COUNT by itself is not equal to the windows from checkpoint. Checkpoint = Window greater than or equal to multiple of DAG checkpoint count that is also a multiple of operator checkpoint window, hence the complication. Also as Tim pointed out my logic has an issue and I will correct it.